stdlib/in_channel.mli

```diff
@@ -0,0 +1,148 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*             Xavier Leroy, projet Cristal, INRIA Rocquencourt           *)
+(*                                                                        *)
+(*   Copyright 2021 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** Input channels.
+
+    @since 4.14.0 *)
+
+type t = in_channel
+(** The type of input channel. *)
+
+type open_flag = Stdlib.open_flag =
+  | Open_rdonly      (** open for reading. *)
+  | Open_wronly      (** open for writing. *)
+  | Open_append      (** open for appending: always write at end of file. *)
+  | Open_creat       (** create the file if it does not exist. *)
+  | Open_trunc       (** empty the file if it already exists. *)
+  | Open_excl        (** fail if Open_creat and the file already exists. *)
+  | Open_binary      (** open in binary mode (no conversion). *)
+  | Open_text        (** open in text mode (may perform conversions). *)
+  | Open_nonblock    (** open in non-blocking mode. *)
+(** Opening modes for {!open_gen}. *)
+
+val stdin : t
+(** The standard input for the process. *)
+
+val open_text : string -> t
+(** Open the named file for reading, and return a new input channel on that
+    file, positioned at the beginning of the file. *)
+
+val open_bin : string -> t
+(** Same as {!open_text}, but the file is opened in binary mode, so that no
+    translation takes place during reads. On operating systems that do not
+    distinguish between text mode and binary mode, this function behaves like
+    {!open_text}. *)
```

Comment:
Also I think should be reordered. You don't want the info about the translation happening in `open_text` in `open_bin` you want it in `open_text`. I would also put `open_bin` before `open_text`.

Reply:
No objection here, docstrings amended.